src/utils/cast-strategy.ts

```diff
@@ -0,0 +1,102 @@
+import { applyDefaultValue, IOttomanType, Schema, ValidationError } from '../schema';
+import { CoreType } from '../schema/types';
+
+/**
+ * Cast Strategies
+ * 'keep' | 'drop' | 'throw' | 'defaultOrDrop' | 'defaultOrKeep'
+ *
+ *  @desc
+ *  When trying to cast a value to a given type if it fail Cast Strategy will behave this ways for:
+ * 'keep' -> The original value will be returned
+ * 'drop' -> Return undefined and the object field will be removed.
+ * 'throw' -> Throw execption 'Value couldn't be casted to given type'
+ * 'defaultOrDrop' -> Try to return default value if exists, if no default value was provided for the current field then it will be removed
+ * 'defaultOrDrop' -> Try to return default value if exists, if no default value was provided for the current field then it will keep original value.
+ */
+export enum CAST_STRATEGY {
+  KEEP = 'keep',
+  DROP = 'DROP',
```

Comment:
try to keep the same value pattern, use all in uppercase or all in lowerCamelCase